source/renderer/app/stores/AssetsStore.js

```diff
@@ -50,8 +64,43 @@ export default class AssetsStore extends Store {
   getAssetDetails = (policyId: string, assetName: string): ?Asset =>
     this.details[policyId + assetName];
 
+  @computed get assetSettingsDialogWasOpened(): boolean {
+    return requestGetter(this.getAssetSettingsDialogWasOpenedRequest, false);
+  }
+
   // =================== PRIVATE ==================
 
+  @action _onAssetSettingsOpen = ({ asset }: { asset: WalletSummaryAsset }) => {
+    this.editingsAsset = asset;
+    this.api.localStorage.setAssetSettingsDialogWasOpened();
+    this.getAssetSettingsDialogWasOpenedRequest.execute();
+  };
+
+  @action _onAssetSettingsSubmit = async ({
+    asset,
+    decimals,
+  }: {
+    asset: WalletSummaryAsset,
+    decimals: number,
+  }) => {
+    this.editingsAsset = null;
+    const { fingerprint, policyId, assetName } = asset;
+    const assetDomain = this.getAssetDetails(policyId, assetName);
+    if (assetDomain) {
+      assetDomain.update({
+        decimals,
+      });
+    }
+    this._refreshAssetsData();
+    await this.api.localStorage.setAssetLocalData(fingerprint, {
+      decimals,
+    });
+  };
+
+  @action _onAssetSettingsCancel = () => {
+    this.editingsAsset = null;
```

Comment:
@daniloprates Do we need to reset `getAssetSettingsDialogWasOpenedRequest` request also?

Reply:
@tomislavhoracek we can't do that, otherwise the PopOver in the Wallet Summary will reappear :)

Reply:
@daniloprates oke ✔️